app/models/system.rb

```diff
@@ -20,7 +20,16 @@ class System < ApplicationRecord
   alias_attribute :scc_synced_at, :scc_registered_at
 
   def init
-    self.login ||= System.generate_secure_login
+    self.login ||= if system_token.presence
```

Comment:
I'd suggest to set the login in https://github.com/SUSE/rmt/blob/master/engines/scc_proxy/lib/scc_proxy/engine.rb#L291, then this is encapsulated to public cloud code.